fw/cache.c

```diff
@@ -261,22 +261,27 @@ static DEFINE_PER_CPU(char[RESP_BUF_LEN], g_c_buf);
 static TfwStr g_crlf = { .data = S_CRLF, .len = SLEN(S_CRLF) };
 
 /* Iterate over request URI and Host header to process request key. */
-#define TFW_CACHE_REQ_KEYITER(c, uri_path, host, u_end, h_start, h_end)	\
+#define TFW_CACHE_REQ_KEYITER(c, uri_path, host, u_end, h_start,	\
+			      h_end, u_fin, h_fin)			\
+	u_fin = h_fin = false;						\
+	BUG_ON(TFW_STR_EMPTY(host));					\
```

Comment:
We can have empty Host, i.e. HTTP/1 requires Host header, but a client can just sent `Host:\n`. Anyway, there is no reason to crash kernel: in worst case just WARN() and return `false`. But in general, I see no reason why not to iterate over empty `Host`. @const-t why we have a broken logic in this case?

Reply:
I remove the BUG_ON but not add WARN with returning false, because this macro is used in different places one of it need false and other need -EINVAL return code. We don't check that URI is not empty and it seems that for host we also can not check it. Host cant be empty because we always check it in `__check_authority_correctness` in http.c. I also add such test for Host:\n and cache